tools/i18n/files.js

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+var fs = require('fs');
+var path = require('path');
+
+function findFiles(dir, ext) {
+    var result = [];
+    if (!isDirectory(dir)) {
+        return result;
+    }
+
+    var files = fs.readdirSync(dir);
+
+    var directories = [];
+    // process files then directories
+    files.forEach(function (file) {
+        file = path.join(dir, file);
+        if (isDirectory(file)) {
+            directories.push(file);
+        } else {
+            if (path.extname(file) === '.' + ext) {
+                result.push(file);
+            }
+        }
+
+    });
+
+    directories.forEach(function (dir) {
+        result = result.concat(findFiles(dir, ext));
+    });
+
+    return result;
+}
+
+function isDirectory(path) {
+    return existsSync(path) && fs.statSync(path).isDirectory();
```

Comment:
If I just called `statSync()` I'd have to wrap it in a try/catch anyway, and `existsSync()` does that for me 😄.